Bug - 2693 - Allow Users to Delete Skills

esizer opened the pull request:
Resolves #2693 

## Summary

This updates the `<ExperienceForm>` component to use the `useFieldArray` hook form `react-hook-form` to handle the skills array. Previously, we were manually updating using `setValue` which was causing problems when attempting to remove items.

## Testing

1. Navigate to the frontend directory, `cd frontend`
2. Build the talent search workspace `npm run production --workspace=talentsearch`
3. Navigate to the [Skills and Experiences](http://localhost:8000/en/talent/profile/skills-and-experiences) page.
4. Edit an existing experience
5. Test adding and removing skills
  - Add with the **Add Skills** section
  - Remove skills from
    - Chips under **Skills** attached to this experience
    - List under **Add Skills** section
    - Using the **Remove from experience** button in **Skills in Detail** section

All of the listed ways to remove a skill should remove it from the interface and send the proper mutation variables to execute the sync.

## Note

While working on this PR, I noticed that the `useForm` hook should be avoided when already within a form context. If you are already within that context, we should be using `useFormContext` in-place of that to get the existing form context. `useForm` seems to create a new context  which can cause problems.

vd1992 reviewed (changes requested):
### General 
_Relevant_
Request for keyprops as shown, appears when loading /edit when atleast one skill has been selected

![image](https://user-images.githubusercontent.com/40485260/170375336-0739f0f8-339f-414a-8a44-843c410a6a3b.png)

### Accessibility 
_Out of scope?_
I took screenshots of the Lighthouse report, included
Multiples of these originate from components generated elsewhere, but given this is a page that uses many components it makes it easier to collate issues here. A bit out of scope here, but maybe something to spin-off down the line. Or maybe to be ignored if another tool is added to the Github Workflow. 

Your call really, but I wouldn't worry about it in this issue but just keep what the alerts say in mind. 

![image](https://user-images.githubusercontent.com/40485260/170375492-1bc054ad-e314-4e62-ab68-e3e96665f01a.png)

![image](https://user-images.githubusercontent.com/40485260/170375530-3a3a0f02-c549-45a7-8d20-98d7b0551449.png)

![image](https://user-images.githubusercontent.com/40485260/170375681-53c14bc1-9adf-4c45-89dc-61427ce83f8c.png)

![image](https://user-images.githubusercontent.com/40485260/170375771-e04a4bc4-6494-4801-98df-c78ae7e7aacb.png)

vd1992 commented:
tried to push things through its paces manually, dunno if you want to add any new tests for this, whether they are even needed
so far, looks great, amazing!

- can create experience
- can update experience
- can delete experience
- can delete skills, save, skills adjusted accordingly (issue resolved)
- refresh form holds data

not sure what else to throw at it

esizer commented:
> _Out of scope?_
> I took screenshots of the Lighthouse report, included

These were simple enough to fix outside of the checkbox so I threw them in. Just updated some roles and adjusted the heading levels.

petertgiles commented on frontend/talentsearch/src/js/components/skills/AddedSkills/AddedSkills.tsx:
```diff
@@ -19,25 +19,29 @@ const AddedSkills: React.FunctionComponent<AddedSkillsProps> = ({
+            const handleDismiss = () => onRemoveSkill(skill.id);
+            return (
+              <Chip
+                key={skill?.id}
```
I'm getting a
```
 Warning: Each child in a list should have a unique "key" prop.
Check the render method of `AddedSkills`.
```
on the skill edit page.  Is it because this is possibly null?

esizer replied:
So, this is the annoying bit. I updated this,`skill.id -> skill?.id` because when testing, we generate mock skills and experiences separately. This results in the `id` for skills not matching the `id` for skills attached to experiences.

It was actually breaking because when we pass in the skills that are added, they don't match the available skills and we get a "cannot access property `id` of `undefined`. This breaks the tests, adding the `?.id` throws some weird errors but does not break the tests.

I think we should create a ticket to update the `fakeExperiences` to allow us to pass in valid skills to be used in the generation. Once that is updated, we can revert this back to remove the warning.

petertgiles reviewed (approved):
Oh yeah, this is way nicer.  Nice fix and refactor! 💪

vd1992 reviewed (approved):
Remaining issue to be addressed elsewhere. 
Great work!

petertgiles reviewed (approved):
Looks nice!
